cpp/include/nvtext/tokenize.hpp

```diff
@@ -175,5 +175,42 @@ std::unique_ptr<cudf::column> character_tokenize(
   cudf::strings_column_view const& strings,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
 
+/**
+ * @brief Creates a strings column from a strings column of tokens and an
+ * associated column of row ids.
+ *
+ * Multiple tokens from the input column may be combined into a single row (string)
+ * in the output column. The tokens are concatenated along with the `separator` string
+ * in the order in which they appear in the `row_indices` column.
+ *
+ * @code{.pseudo}
+ * Example:
+ * s = ["hello", "world", "one", "two", "three"]
+ * r = [0, 0, 1, 1, 1]
```

Comment:
What would be the output with indices `[0, 0, 2, 2, 2]` ?

Reply:
Yes, I just tried it and it does crash. I think this can be allowed. I don't want to add _holes_ to the output but I think we can treat the indices as relative so `[0,0,1,1,1]` produces the same output as `[0,0,2,2,2], [10,10,20,20,20]`.
This is all beyond the original requirement so throwing an error should be acceptable as well.

Reply:
@VibhuJawa

Reply:
I can fix the crash. My inclination is to document this as "unexpected data causes undefined behavior" rather than adding pre-checks to the function.

Reply:
So I agree that:

a. This is out of the scope of the initial requirement 

b. This is more of a nice to have to kind of a feature, which may be useful for handling cases like empty docs after pre-processing 

c. A user can do below in python to work around it for now. 

```python
import cudf
s = cudf.Series(["hello", "world", "one", "two", "three"])
r = cudf.Series([0, 0, 2, 2, 2],name='doc_id')

r_unique_df = r.unique().reset_index(drop=False)
r_increasing_df = r_unique_df.merge(r)
del r
print(r_increasing_df)
```
```
   index  doc_id
0      0       0
1      0       0
2      1       2
3      1       2
4      1       2
```

FWIW, I am totally fine with making the end-user responsible for this for now. 


CC: @beckernick / @randerzander , In case they have thoughts on this.

Reply:
I fixed the crash and added comment to the doxygen:

>  The values in `row_indices` are expected to have positive, sequential values without any missing row indices otherwise the output is undefined.